lib/samly/sp_handler.ex

```diff
@@ -156,14 +172,21 @@ defmodule Samly.SPHandler do
   # non-ui logout request from IDP
   def handle_logout_request(conn) do
     %IdpData{id: idp_id} = idp = conn.private[:samly_idp]
-    %IdpData{esaml_idp_rec: idp_rec, esaml_sp_rec: sp_rec} = idp
+
+    %IdpData{
+      post_session_cleanup_pipeline: pipeline,
+      esaml_idp_rec: idp_rec,
+      esaml_sp_rec: sp_rec
+    } = idp
+
     sp = ensure_sp_uris_set(sp_rec, conn)
 
     saml_encoding = conn.body_params["SAMLEncoding"]
     saml_request = conn.body_params["SAMLRequest"]
     relay_state = conn.body_params["RelayState"] |> safe_decode_www_form()
 
-    with {:ok, payload} <- Helper.decode_idp_signout_req(sp, saml_encoding, saml_request) do
+    with {:ok, payload} <- Helper.decode_idp_signout_req(sp, saml_encoding, saml_request),
+         {:halted, %Conn{halted: false} = conn} <- {:halted, pipethrough(conn, pipeline)} do
```

Comment:
Is it correct that the same pipeline is invoked for both the logout request and the logout response?

Reply:
Correct, in my implementation that would be the case. But both of them will not happen as here the code pathway is distinguishing between an IDP initiated request and the response after we initiate a request as a SP.

https://github.com/workera-ai/elixir_samly/blob/5f4c507ab973877955c392d957b0501239effe42/lib/samly/sp_router.ex#L22-L28